StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1501UnitTests.cs

```diff
@@ -490,15 +512,339 @@ void MethodName()
+            Debug.Assert(true);
+else
+            Debug.Assert(false);
+    }
+}";
+
+            await this.VerifyCSharpFixAsync(testCode, fixedTestCode, cancellationToken: CancellationToken.None).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Verifies that the code fix provider will properly handle alternate indentations.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task TestNoSA1503CodeFixProviderWithAlternateIndentationAsync()
+        {
+            this.IndentationSize = 1;
+            this.suppressSA1503 = true;
+
+            var testCode = @"using System.Diagnostics;
+public class Foo
+{
+ public void Bar(int i)
+ {
+  if (i == 0) Debug.Assert(true);
+ }
+}";
+
+            var fixedTestCode = @"using System.Diagnostics;
+public class Foo
+{
+ public void Bar(int i)
+ {
+  if (i == 0)
+   Debug.Assert(true);
+ }
+}";
+
+            await this.VerifyCSharpFixAsync(testCode, fixedTestCode).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Verifies that the code fix provider will work properly handle non-whitespace trivia.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task TestNoSA1503CodeFixProviderWithNonWhitespaceTriviaAsync()
+        {
+            this.suppressSA1503 = true;
+
+            var testCode = @"using System.Diagnostics;
+public class Foo
+{
+    public void Bar(int i)
+    {
+#pragma warning restore
+        if (i == 0)
+            Debug.Assert(true);
+    }
+}";
+
+            var fixedTestCode = @"using System.Diagnostics;
+public class Foo
+{
+    public void Bar(int i)
+    {
+#pragma warning restore
+        if (i == 0)
+            Debug.Assert(true);
+    }
+}";
+
+            await this.VerifyCSharpFixAsync(testCode, fixedTestCode).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Verifies that the code fix provider will work properly handle multiple cases of missing brackets.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task TestNoSA1503CodeFixProviderWithMultipleNestingsAsync()
+        {
+            this.suppressSA1503 = true;
+
+            var testCode = @"using System.Diagnostics;
+public class Foo
+{
+    public void Bar(int i)
+    {
+        if (i == 0) if (i == 0) Debug.Assert(true);
+    }
+}";
+
+            var fixedTestCode = @"using System.Diagnostics;
+public class Foo
+{
+    public void Bar(int i)
+    {
+        if (i == 0)
+            if (i == 0)
+                Debug.Assert(true);
+    }
+}";
+
+            var batchFixedTestCode = @"using System.Diagnostics;
+public class Foo
+{
+    public void Bar(int i)
+    {
+        if (i == 0)
+            if (i == 0)
+    Debug.Assert(true);
```

Comment:
:arrow_right: I fixed this by only reporting (and thus fixing) the first instance of SA1501 on a line (still for the specific case where SA1503 is disabled).